### PR TITLE
SWC-6645

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/presenter/SearchPresenter.java [hunk 1]
```diff
@@ -89,6 +91,15 @@ public void setPlace(Search place) {
     String queryTerm = place.getSearchTerm();
     if (queryTerm == null) queryTerm = "";
     currentSearch = checkForJson(queryTerm);
+    if (currentSearch.getQueryTerm().size() == 1) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/SearchPresenter.java:94]: If there's a single search term, check to see if we should redirect (to Synapse entity page or People Search)

File: src/main/java/org/sagebionetworks/web/client/presenter/SearchUtil.java [hunk 2]
```diff
@@ -34,6 +34,9 @@ public static Place willRedirect(Search place) {
    * @return
    */
   public static Place willRedirect(String queryTerm) {
+    if (queryTerm == null || queryTerm.trim().length() == 0) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/SearchUtil.java:37]: Precondition check

File: src/main/java/org/sagebionetworks/web/client/presenter/TeamSearchPresenter.java [hunk 3]
```diff
@@ -79,6 +79,11 @@ public void setPlace(TeamSearch place) {
     this.place = place;
     this.view.setPresenter(this);
     searchTerm = place.getSearchTerm();
+    Place redirectPlace = SearchUtil.willRedirect(searchTerm);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/TeamSearchPresenter.java:82]: Do the same for Team Search

File: src/main/java/org/sagebionetworks/web/client/presenter/TeamSearchPresenter.java [hunk 4]
```diff
@@ -98,25 +103,26 @@ public void goTo(Place place) {
   public void loadMore() {
     synAlert.clear();
     // execute search, and update view with the results
-    AsyncCallback<PaginatedResults<Team>> callback = new AsyncCallback<PaginatedResults<Team>>() {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/TeamSearchPresenter.java:101]: No code change, auto-formatted on update

**Review by nickgros — APPROVED** [review]
Looks good!

**Pull request merged**